### ISSUE TITLE
memory leak when call gocv.split()

### ISSUE THREAD

**Issue by HsuTzuJen** [hybridgroup/gocv#394]
## Description

example:
Assum that img is a 2 channel Mat

for {
    imgarr := gocv.splite(img)
    imgarr[0].Close()
    imgarr[1].Close()
}
Kind of above codes cause memory leak. I am sure that the Mat objects in imgarr are closed, but the memory usage is still growing up. Why?

## Steps to Reproduce

1. https://stackoverflow.com/questions/54456414/garbage-collection-is-not-working-with-gocv-mat-in-golang/54457054#54457054
2.
3.
4.

## Your Environment

* Operating System and version:win10
* OpenCV version used:4.0
* How did you install OpenCV?as official site instructions
* GoCV version used:4.0
* Go version:1.11.5
* Did you run the `env.sh` or `env.cmd` script before trying to `go run` or `go build`?no

**Comment by dougnd** [comment]
I can't seem to reproduce a leak.  

The following code does not seem to use a noticeably larger amount of memory:
```
import (
	"log"
	"time"

	"gocv.io/x/gocv"
)

func main() {
	img := gocv.NewMatWithSizeFromScalar(gocv.Scalar{0, 0, 0, 0}, 500, 500, gocv.MatTypeCV8UC3)
	log.Print("waiting...")
	<-time.After(5 * time.Second)
	log.Print("looping...")
	for i := 0; i < 1000000; i++ {
		matsplits := gocv.Split(img)
		matsplits[0].Close()
		matsplits[1].Close()
		matsplits[2].Close()
	}
	log.Print("waiting...")
	<-time.After(10 * time.Second)
}
```

How many loops does it take for the memory usage to be noticeable?

That said, I don't see where [this array allocation](https://github.com/hybridgroup/gocv/blob/66a9ff38e8e0fafd14a0d12ae6b6169f7bbba36f/core.cpp#L572) is freed.  Perhaps there should be a call to `C.Mats_Close` in  [Split](https://github.com/hybridgroup/gocv/blob/66a9ff38e8e0fafd14a0d12ae6b6169f7bbba36f/core.go#L1542)?

**Comment by linyongzuo** [comment]
i think you should try this
`
length := len(imgarr)
for i := 0; i < length; i ++ {
imgarr[i].Close()
}
`

**Comment by deadprogram** [comment]
This was corrected by #580 now closing. Thanks!
